pkg/ingestr/operator.go

```diff
@@ -271,21 +154,21 @@ func (o *BasicOperator) ConvertTaskInstanceToContainerConfig(ctx context.Context
 		cmdArgs = append(cmdArgs, "--full-refresh")
 	}
 
-	if len(mounts) == 0 {
-		mounts = nil
+	path := ti.GetAsset().ExecutableFile.Path
+	repo, err := o.finder.Repo(path)
+	if err != nil {
+		return errors.Wrap(err, "failed to find repo to run Ingestr")
+	}
+
+	if strings.HasPrefix(destURI, "duckdb://") {
+		duck.LockDatabase(destURI)
+		defer duck.UnlockDatabase(destURI)
+	}
+
+	if strings.HasPrefix(sourceURI, "duckdb://") {
+		duck.LockDatabase(sourceURI)
+		defer duck.UnlockDatabase(sourceURI)
 	}
```

Comment:
this could get into a locked situation if the user is moving data within the same db?